akka-actor-typed/src/main/scala/akka/actor/typed/internal/delivery/DurableProducerState.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed.internal.delivery
+
+import akka.actor.typed.ActorRef
+
+object DurableProducerState {
```

Comment:
I have added a first sketch of how to add the optionally durable part of the ProducerController. See https://github.com/akka/akka/pull/28155/commits/4afad2f17c1116693a7ecefb0bc8480e00f55bbd

Not tested at all yet, because first I would like feedback on the direction.

This `DurableProducerState` is supposed to be and actor with below message protocol. We provide an implementation with `EventSourcedBehavior`.

The `ProducerController` asks StoreMessageSent before sending the message, and `StoreMessageConfirmed` when confirmation is received.